docs/content/any/reference/polar/polar-syntax.md

```diff
@@ -586,3 +605,268 @@ To add an inline query to a Polar file, use the `?=` operator:
 ```
 
 An inline query is only valid at the beginning of a line.
+
+### Rule Types
+
+A rule type is simply the declaration of a rule that specifies the rule's name, arguments and argument types or values.
+Rule types have the same syntax as rule head, and are preceded by the keyword `type`.
+
+For example,
+
+```polar
+type has_permission(actor: Actor, action: String, resource: Resource);
+```
+
+The above rule type specifies that any rule with the name `has_permission` must have three arguments, where the first argument matches `Actor`, the second argument matches `String`, and the third argument matches `Resource`.
+
+Argument matching is determined in the same way that matching is determined for rule evaluation. See [Patterns and Matching](#patterns-and-matching).
+
+Rule types are optional. If a rule type exists with the same name as a rule, then the rule must match that type or else an error will be thrown at policy load-time.
+
+You can find a reference for built-in rule types [here](reference/polar/builtin_rule_types).
+
+## Actor and Resource Blocks
```

Comment:
Everything below here could be broken into a separate page specifically about blocks, which was the original plan, and we could link to that page from a brief overview here.